docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,118 @@
+- Environmental variables can override command line options (except the -o overwrite and -r registry command line options). 
+
+NOTES:
+- Environmental variables do not override any local configuration; that is when configuration for a service is obtained from the local config file but not from the configuration service, the environmental variables are ignored.
+- Environmental variable overrides remove the need to change the "docker" profile in the res/docker/configuration.toml files - Allowing removal of 50% of the existing configuration.toml files.
+
+All values overriden get logged (indicating which configuration value or op param and the new value).  
+
+## Decision 
+
+These features have been implemented (with some minor changes to be done) for consideration here:  https://github.com/edgexfoundry/go-mod-bootstrap/compare/master...lenny-intel:SelfSeed2
+
+The implementation for self-seeding services and environmental overrides is already implemented (for Fuji) per this document in the application services and device services (and instituted in the SDKs of each).
+
+## Backward compatibility
+Several aspects of this ADR contain backward compatibility issues for the device service and application service SDKs.  Therefore, for the upcoming minor release, the following guidelines and expections are added to provide for backward compatibility.
+
+- --registry=<url> for Device SDK
+        
+If used with URL , assume previous version and use URL for --configProvider option and default to http as DS SDK does today. Also use URL for registry connection.
+
+- --registry (no ‘=’) and w/o --configProvider for both SDKs
+
+Assume for this release that one would never use registry w/o config provider.
+Use registry configuration from file for config provider as is done in Fuji. 
+
+- Env Var: edgex_registry=<url> for all services (currently has been removed)
+
+Add it back and use value as if it was edgex_configuration_provider and enable use of registry with same settings in URL. Default to http as it is in Fuji.
+ 
+## Consequences 
+
+- Docker compose files will need to be changed to remove config seed. 
+- The main Snap will need to be changed to remove config seed. 
+- Config seed code (currently in edgex-go repo) is to be removed.
+- Any service specific environmental overrides currently on config seed need to be moved to the specific service(s).
+- The Docker configuration files and directory (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml) that are used to populate the config seed for Docker containers can be eliminated from all the services.
```

Comment:
In at least one case (security-file-token-provider under cmd/security-secretstore-setup) there's only a docker configuration.toml.

Reply:
Good catch, that file could be moved rather than deleted.